libchisel/src/checkstartfunc.rs

```diff
@@ -14,6 +14,25 @@ impl CheckStartFunc {
     }
 }
 
+use std::collections::HashMap;
+
+impl ModuleConfig for CheckStartFunc {
+    fn with_defaults() -> Result<Self, ModuleError> {
+        Err(ModuleError::NotSupported)
+    }
+
+    fn with_config(config: &HashMap<String, String>) -> Result<Self, ModuleError> {
```

Comment:
Could also split this into two (and that would remove the need for default):
- "with config string"
- "with complex config"

e.g.
```
fn with_config_string(config: &str)
fn with_config_map(config: &HashMap<String, String>)
```

and then `with_config_string(&"")` would mean "defaults"

Reply:
I think key-val configuration should be fine for now. No need to further complicate things with the config string, and we can definitely just implement the config string as a specified key `config` or something in the hashmap.

Reply:
True, but the downside is the terrible syntax needed for creating a hashmap and passing it. So for simple modules a string is much more simple.

Reply:
Perhaps the config string method could just wrap that logic?

Reply:
What do you mean by wrap? Define a way to encode hashmaps?

Reply:
`with_config_string` would just call `with_config` where a dedicated key such as `config_string` is set to the value provided.